src/connector/src/sink/console.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use itertools::join;
+use risingwave_common::array::{Op, StreamChunk};
+use risingwave_common::catalog::Schema;
+use risingwave_common::types::{DatumRef, ScalarRefImpl};
+
+use crate::sink::{Result, Sink};
+
+pub const CONSOLE_SINK: &str = "console";
+
+#[derive(Clone, Debug)]
+pub struct ConsoleConfig {
+    pub prefix: Option<String>,
+    pub suffix: Option<String>,
+}
+
+impl ConsoleConfig {
+    pub fn from_hashmap(values: HashMap<String, String>) -> Result<Self> {
+        Ok(ConsoleConfig {
+            prefix: values.get("prefix").cloned(),
+            suffix: values.get("suffix").cloned(),
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct ConsoleSink {
+    pub epoch: u64,
+    pub buffer: Vec<String>,
+    pub prefix: String,
+    pub suffix: String,
+}
+
+impl ConsoleSink {
+    pub async fn new(config: ConsoleConfig) -> Result<Self> {
+        Ok(ConsoleSink {
+            epoch: 0,
+            buffer: vec![],
+            prefix: config.prefix.unwrap_or("".to_string()),
+            suffix: config.suffix.unwrap_or("".to_string()),
+        })
+    }
+
+    fn parse_datum(datum: DatumRef<'_>) -> String {
```

Comment:
This is not a method specially for `ConsoleSink`. Maybe we can move it to outside `impl ConsoleSink`.